### ISSUE TITLE
Unable to get it working on my python module

### ISSUE THREAD

**Issue by harshitpatel96** [Aluriak/clyngor#21]
This is the error I get while using it. The have_clingo_module is infact present in the clyngor library but still python pops this error out. Besides there are other bugs like Raise(error) command has string written outside of inverted commas, and this gives error when I initally tried to import clyngor. I manually corrected each raise error.

```
Traceback (most recent call last):

  File "<ipython-input-19-705a45273486>", line 7, in <module>
    """)

  File "C:\Users\HARSHIT\Anaconda3\lib\site-packages\clyngor\inline.py", line 17, in ASP
    return solve(inline=source_code, stats=False, **kwargs)

  File "C:\Users\HARSHIT\Anaconda3\lib\site-packages\clyngor\solving.py", line 66, in solve
    use_clingo_module = use_clingo_module and clyngor.have_clingo_module() and not return_raw_output

AttributeError: module 'clyngor' has no attribute 'have_clingo_module'
```

**Comment by Aluriak** [comment]
Can you provide information about your OS, python version, packaging system (anaconda i guess ?), and the minimal working python code that produces the error ?

> Besides there are other bugs […] I manually corrected each raise error.

Can you show them, or PR them ?

**Comment by harshitpatel96** [comment]
Sure, here are the syntax errors that I am getting when I install clyngor and import it.

`import clyngor
Traceback (most recent call last):

  File "C:\Users\HARSHIT\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py", line 2862, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-82-81dda994ca12>", line 1, in <module>
    import clyngor

  File "D:\NCSU\Semester_1\CSC503 - Computational Applied Logic\Project\clyngor\__init__.py", line 4, in <module>
    from clyngor.utils import ASPSyntaxError, ASPWarning, parse_clingo_output, clingo_value_to_python, with_clingo_bin, opt_models_from_clyngor_answers, answer_set_to_str, answer_set_from_str

  File "D:\NCSU\Semester_1\CSC503 - Computational Applied Logic\Project\clyngor\utils.py", line 147
    raise ValueError(f"Python value '{repr(val)}' of type {type(val)} is not convertible in ASP.")
                                                                                                ^
SyntaxError: invalid syntax`

**Comment by harshitpatel96** [comment]
Sure, here are the syntax errors that I am getting when I install clyngor and import it.

`import clyngor
Traceback (most recent call last):

File "C:\Users\HARSHIT\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py", line 2862, in run_code
exec(code_obj, self.user_global_ns, self.user_ns)

File "", line 1, in
import clyngor

File "D:\NCSU\Semester_1\CSC503 - Computational Applied Logic\Project\clyngor_init_.py", line 4, in
from clyngor.utils import ASPSyntaxError, ASPWarning, parse_clingo_output, clingo_value_to_python, with_clingo_bin, opt_models_from_clyngor_answers, answer_set_to_str, answer_set_from_str

File "D:\NCSU\Semester_1\CSC503 - Computational Applied Logic\Project\clyngor\utils.py", line 147
raise ValueError(f"Python value '{repr(val)}' of type {type(val)} is not convertible in ASP.")
^
SyntaxError: invalid syntax`
 
I just removed f from the raise ValueError line and then I was able to import it. But later I got the error that I mentioned in my initial query. I was not able to correct that.

**Comment by harshitpatel96** [comment]
Hey,
I accidentally closed the post, if you can undo it then please do it and
resolve the issue.

Thanks & Regards,
Harshit Patel

**Comment by harshitpatel96** [comment]
Never mind, I reopened it

Thanks & Regards,
Harshit Patel

**Comment by Aluriak** [comment]
Sounds like your installation does not support f-string, [introduced in python 3.6](https://www.python.org/dev/peps/pep-0498/).

Since i tried to get clyngor work for lesser versions, i should fix that, thanks for pointing that up.
Until then, update your python installation or remove all f-strings.

**Comment by Aluriak** [comment]
f-string fixed since a7b05ecdbd5f73dbee5a7244a96995d9035f3695.

**Comment by Aluriak** [comment]
Can you test again for that bug, and if it's fixed, closing that issue ?
